mqtt-client-Node.js/index.js

```diff
@@ -1,71 +1,122 @@
+ *
+ * for more details about "mqtt.connect" method & options,
+ * please refer to https://github.com/mqttjs/MQTT.js#mqttconnecturl-options
+ */
+let connectUrl = `${protocol}://${host}:${port}`
+if (['ws', 'wss'].includes(protocol)) {
+  // mqtt: MQTT-WebSocket uniformly uses /path as the connection path,
+  // which should be specified when connecting, and the path used on EMQX is /mqtt.
+  connectUrl += '/mqtt'
+}
+if (['mqtts', 'wss'].includes(protocol)) {
+  options['ca'] = fs.readFileSync('./broker.emqx.io-ca.crt')
+}
 
-const topic = '/nodejs/mqtt'
+const client = mqtt.connect(connectUrl, options)
 
-const client = mqtt.connect(connectUrl, OPTIONS)
+const topic = '/nodejs/mqtt'
+const payload = 'nodejs mqtt test'
+// https://github.com/mqttjs/MQTT.js#qos
+const qos = 0
 
+// https://github.com/mqttjs/MQTT.js#event-connect
 client.on('connect', () => {
-  console.log(`${program.protocol}: Connected`)
-  client.subscribe([topic], () => {
-    console.log(`${program.protocol}: Subscribe to topic '${topic}'`)
+  console.log(`${protocol}: Connected`)
+
+  // subscribe topic
+  // https://github.com/mqttjs/MQTT.js#mqttclientsubscribetopictopic-arraytopic-object-options-callback
+  client.subscribe(topic, { qos }, () => {
+    console.log(`${protocol}: Subscribe to topic '${topic}'`)
   })
-  client.publish(topic, 'nodejs mqtt test', { qos: 0, retain: false }, (error) => {
+
+  // publish message
+  // https://github.com/mqttjs/MQTT.js#mqttclientpublishtopic-message-options-callback
+  client.publish(topic, payload, { qos }, (error) => {
     if (error) {
       console.error(error)
     }
   })
 })
 
+// https://github.com/mqttjs/MQTT.js#event-reconnect
 client.on('reconnect', (error) => {
-  console.log(`Reconnecting(${program.protocol}):`, error)
+  console.log(`Reconnecting(${protocol}):`, error)
 })
 
+// https://github.com/mqttjs/MQTT.js#event-error
 client.on('error', (error) => {
-  console.log(`Cannot connect(${program.protocol}):`, error)
+  console.log(`Cannot connect(${protocol}):`, error)
 })
 
+// https://github.com/mqttjs/MQTT.js#event-message
 client.on('message', (topic, payload) => {
   console.log('Received Message:', topic, payload.toString())
 })
+
+// unsubscribe after 10 seconds
+setTimeout(() => {
+  // unsubscribe topic
+  // https://github.com/mqttjs/MQTT.js#mqttclientunsubscribetopictopic-array-options-callback
+  client.unsubscribe(topic, { qos }, (error) => {
+    if (error) {
+      console.log('unsubscribe error:', error)
+      return
+    }
+    console.log(`unsubscribed topic: ${topic}`)
+  })
+}, 10000)
+
+// disconnect after 12 seconds
+setTimeout(() => {
+  if (client.connected) {
+    try {
+      // disconnect
+      // https://github.com/mqttjs/MQTT.js#mqttclientendforce-options-callback
+      client.end(false, () => {
+        console.log('disconnected successfully')
+      })
+    } catch (error) {
+      btnLoadingType.value = ''
+      console.log('disconnect error:', error)
+    }
+  }
+}, 12000)
```

Comment:
I have no idea why it's necessary to implement automatic disconnection and unsubscription here.

Reply:
I noticed that the previous few demos all implemented disconnection and unsubscription.

Reply:
I don't know how to manually disconnect and unsubscribe here, so I used setTimeout.

Reply:
Okay, I see your point, but this is a demo used in Node.js, usually not an application, unlike some other examples, users need to edit the code directly to complete, not use manual commands unless you package it as a CLI similar to MQTTX CLI.

However, users will likely be confused when running it, if not needed by the user, you can comment out this code and let the user modify it according to their needs, instead of automatically unsubscribing and disconnecting after the timeout.

Reply:
Okay, I'll comment out this code and add some notes.